sdk/keyvault/azure-security-keyvault/azure/security/keyvault/secrets/README.md

```diff
@@ -0,0 +1,231 @@
+
+    print(updated_secret.name)
+    print(updated_secret.value)
+    print(updated_secret.version)
+    print(updated_secret.updated)
+    print(updated_secret.content_type)
+    print(updated_secret.tags)
+
+```
+
+### Delete a Secret
+`delete_secret` deletes a secret previously stored in the Key Vault. When [soft-delete](https://docs.microsoft.com/en-us/azure/key-vault/key-vault-ovw-soft-delete) is not enabled for the Key Vault, this operation permanently deletes the secret.
+```python
+    secret = secret_client.delete_secret("secret-name")
+
+    print(deleted_secret.name)
+    print(deleted_secret.deleted_date)
+```
+### List secrets
+This example lists all the secrets in the specified Key Vault.
+```python
+    secrets = secret_client.list_secrets()
+
+    for secret in secrets:
+        # the list doesn't include values or versions of the secrets
+        print(secret.name)
+```
+
+### Async operations
+Python’s [asyncio package](https://pypi.org/project/asyncio/) (introduced in Python 3.4) and its two keywords `async` and `await` serves to declare, build, execute, and manage asynchronous code.
+The following examples provide code snippets for performing async operations in the Secret Client library:
+
+### Async create a secret
+This example creates a secret in the specified Key Vault with the specified optional arguments.
+```python
+    import asyncio
+
+    from azure.identity import AsyncDefaultAzureCredential
+    from azure.security.keyvault.aio import SecretClient
+
+    credential = AsyncDefaultAzureCredential()
+    # Create a new secret client using a client secret credential
+    secret_client = SecretClient(vault_url=vault_url, credential=credential)
+
+    # Sends secret data and asynchronously waits until acknowledgement is received
+    secret = await secret_client.set_secret("secret-name", "secret-value", enabled=True)
+
+    print(secret.name)
+    print(secret.value)
+    print(secret.version)
+    print(secret.enabled)
+```
+### Async list secrets
+This example lists all the secrets in the specified Key Vault.
+```python
+    secrets = secret_client.list_secrets()
+
+    # Recieve secrets data asynchronously from Key Vault service
+    async for secret in secrets:
+        # the list doesn't include values or versions of the secrets
+        print(secret.name)
+```
+
+## Troubleshooting
+### General
+The Key Vault APIs generate exceptions that can fall into one of the azure-core defined exceptions. For more detailed infromation about exceptions and how to deal with them, see [Azure Core exceptions](TODO).
+
+For example, if you try to retrieve a secret after it is deleted a `404` error is returned, indicating resource not found. In the following snippet, the error is handled gracefully by catching the exception and displaying additional information about the error.
+```python
+try:
+    secret_client.get_secret("deleted_secret")
+except ResourceNotFoundError as e:
+    print(e.message)
+    raise
```

Comment:
🤔  that's propagating, not handling, the exception

Reply:
how should we write a better way of handling it?

Reply:
Actually handle the exception. That means preventing it propagating down the stack, i.e. doing anything but re-raising it.

Reply:
Didn't realize, it was still being raised.
fixed it!